src/components/periodic-report/handlers/sync-periodic-report-to-project.handler.ts

```diff
@@ -34,82 +31,67 @@ export class SyncPeriodicReportToProject
     const project = event.updated;
     const previous = event.previous;
 
+    // Check if the project has valid date range set and changed from the previous set
     const isDateRangeChanged =
-      project.mouStart !== previous.mouStart ||
-      project.mouEnd !== previous.mouEnd;
+      project.mouStart.value &&
+      project.mouEnd.value &&
+      (project.mouStart.value.toMillis() !==
+        previous.mouStart.value?.toMillis() ||
+        project.mouEnd.value.toMillis() !== previous.mouEnd.value?.toMillis());
 
     if (isDateRangeChanged) {
-      await this.removeOldReports(project, event.session);
-      await this.syncRecords(project, event.session);
+      await this.periodicReports.removeFinancialReports(project, event.session);
+      await this.periodicReports.removeNarrativeReports(project, event.session);
+      await this.addRecords(project, event.session);
```

Comment:
I don't think this is exactly right. I'd like to diff to determine the additions / removals. That way we aren't deleting and recreating a bunch of reports for the same time range.
i.e.
01-01-2020 - 01-01-2024
to
01-01-2020 - 01-01-2025
should only add 4 reports. Instead of deleting 16 & creating 20.